### PR TITLE
feat: Kubeflow on PCI

### PR DESCRIPTION
**Pull request by sguyennet**

### CODE REVIEW

**Comment by philippart-s** [comment]
👋 @scraly 👀

**Comment by scraly** [comment]
Thanks for the notification, yes it's already in my Todo list 😊😊😊

File: use-cases/kubeflow/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+## Customize the deployment
+
+Configure Terraform with the public cloud project ID:
+
+```bash
+vim terraform.tfvars
+```
+
+```bash
+ovh_os_project_id = <your_openstack_project_id>
```
Comment (scraly) [inline use-cases/kubeflow/README.md:67]: If you think about public cloud ID/service_name,
instead, we already defined OVH_CLOUD_PROJECT_SERVICE, no? :)

It's better to use only one variable if the content is the same and maybe name it in a better understable way: ovh_cloudproject_service?

File: use-cases/kubeflow/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+```bash
+export OVH_ENDPOINT="ovh-eu"
+export OVH_BASEURL="https://eu.api.ovh.com/1.0/"
+export OVH_APPLICATION_KEY="<your_application_key>"
+export OVH_APPLICATION_SECRET="<your_application_secret>"
+export OVH_CONSUMER_KEY="<your_consumer_key>"
+export OVH_CLOUD_PROJECT_SERVICE="$OS_TENANT_ID"
+
+export TF_VAR_ovh_api_dns_application_key="<your_application_key>"
```
[the remark this answers is not in the document]
Comment (scraly) [reply use-cases/kubeflow/README.md:50]: Moreover, why we need another variable name? You can't use the original one we defined?

File: use-cases/kubeflow/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+export TF_VAR_ovh_api_dns_application_secret="<your_application_secret>"
```
[the remark this answers is not in the document]
Comment (scraly) [reply use-cases/kubeflow/README.md:51]: Moreover, why we need another variable name? You can't use the original one we defined?

File: use-cases/kubeflow/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+export TF_VAR_ovh_api_dns_consumer_key="<your_consumer_key>"
```
[the remark this answers is not in the document]
Comment (scraly) [reply use-cases/kubeflow/README.md:52]: Moreover, why we need another variable name? You can't use the original one we defined?

File: use-cases/kubeflow/provider.tf [hunk 5]
```diff
@@ -0,0 +1,63 @@
+terraform {
+  required_providers {
+    ovh = {
+      source  = "ovh/ovh"
+      version = "0.37.0"
```
Comment (scraly) [inline use-cases/kubeflow/provider.tf:5]: Can you test with 1.0.0? :)

**Comment by scraly** [comment]
Moreover, what about giving at the end of the README, a "hello world" for KubeFlow? A starting point to use our newly deployed kubeflow? :)

**Comment by scraly** [comment]
And in the provider.tf file, in the aws provider section, I think you just have to defined creds and use high perf bucket instead of old one:

```
 endpoints = {
        s3 = "https://s3.<region_in_lowercase>.perf.cloud.ovh.net/"
      }   
```

```
# The following fields should be added if your S3 user credentials are not
      # already configured in files ~/.aws/credentials, ~/.aws/config or in
      # environment variables.
      access_key                  = "s3 user access key"
      secret_key                  = "s3 user secret key"  
 ```
 
 WDYT?

File: use-cases/kubeflow/README.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+## Deploy the stack
+
+```bash
+source ovhrc.sh
+terraform init
+terraform plan
+terraform apply
+```
+
+## Get the Kubeflow default user password
```
Comment (scraly) [inline use-cases/kubeflow/README.md:91]:
```suggestion
## Access to the Kubeflow UI
```

File: use-cases/kubeflow/README.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+```bash
```
Comment (scraly) [inline use-cases/kubeflow/README.md:93]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+terraform output kubeflow_password
```
Comment (scraly) [inline use-cases/kubeflow/README.md:94]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+```
```
Comment (scraly) [inline use-cases/kubeflow/README.md:95]:
```suggestion
Get the Kubeflow URL:

```bash
KUBEFLOW_URL=$(terraform output kubeflow_url)
echo $KUBEFLOW_URL
```

Then the user and the password:

KUBEFLOW_USER=$(terraform output kubeflow_user)
KUBEFLOW_PASSWORD=$(terraform output kubeflow_password)
echo $KUBEFLOW_USER
echo $KUBEFLOW_PASSWORD

Then access to the KubeFlow URL with the user and password.
```

**Review by scraly — CHANGES_REQUESTED** [review]

File: use-cases/kubeflow/helm_external_dns.tf [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+  set {
+    name            = "env[1].value"
+    value           = "${var.ovh_api_dns_application_secret}"
+  }
+
+   set {
+    name            = "env[2].name"
+    value           = "OVH_CONSUMER_KEY"
+  }
```
Comment (scraly) [inline use-cases/kubeflow/helm_external_dns.tf:84]:
```suggestion
```

File: use-cases/kubeflow/helm_external_dns.tf [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+  set {
```
Comment (scraly) [inline use-cases/kubeflow/helm_external_dns.tf:86]:
```suggestion
```

File: use-cases/kubeflow/helm_external_dns.tf [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+    name            = "env[2].value"
```
Comment (scraly) [inline use-cases/kubeflow/helm_external_dns.tf:87]:
```suggestion
```

File: use-cases/kubeflow/helm_external_dns.tf [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+    value           = "${var.ovh_api_dns_consumer_key}"
```
Comment (scraly) [inline use-cases/kubeflow/helm_external_dns.tf:88]:
```suggestion
```

File: use-cases/kubeflow/helm_external_dns.tf [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+  }
```
Comment (scraly) [inline use-cases/kubeflow/helm_external_dns.tf:89]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+export OVH_APPLICATION_KEY="<your_application_key>"
+export OVH_APPLICATION_SECRET="<your_application_secret>"
+export OVH_CONSUMER_KEY="<your_consumer_key>"
+export OVH_CLOUD_PROJECT_SERVICE="<your_public_cloud_project_ID>"
+```
+
+You should create a second OVHcloud credential specific for the DNS configuration with limited permissions:
+https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/ovh.md#creating-ovh-credentials
+
+```bash
```
Comment (scraly) [inline use-cases/kubeflow/README.md:53]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+vim ovhrc.sh
```
Comment (scraly) [inline use-cases/kubeflow/README.md:54]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+```
```
Comment (scraly) [inline use-cases/kubeflow/README.md:55]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+
+```bash
```
Comment (scraly) [inline use-cases/kubeflow/README.md:57]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+export TF_VAR_ovh_api_dns_application_key="<your_dns_application_key>"
```
Comment (scraly) [inline use-cases/kubeflow/README.md:58]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+export TF_VAR_ovh_api_dns_application_secret="<your_dns_application_secret>"
```
Comment (scraly) [inline use-cases/kubeflow/README.md:59]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 21]
```diff
@@ -0,0 +1,123 @@
...
+export TF_VAR_ovh_api_dns_consumer_key="<your_dns_consumer_key>"
```
Comment (scraly) [inline use-cases/kubeflow/README.md:60]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+```
```
Comment (scraly) [inline use-cases/kubeflow/README.md:61]:
```suggestion
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Comment by sguyennet** [comment]
> Moreover, what about giving at the end of the README, a "hello world" for KubeFlow? A starting point to use our newly deployed kubeflow? :)

Done

**Comment by sguyennet** [comment]
> And in the provider.tf file, in the aws provider section, I think you just have to defined creds and use high perf bucket instead of old one:
> 
> ```
>  endpoints = {
>         s3 = "https://s3.<region_in_lowercase>.perf.cloud.ovh.net/"
>       }   
> ```
> 
> ```
> # The following fields should be added if your S3 user credentials are not
>       # already configured in files ~/.aws/credentials, ~/.aws/config or in
>       # environment variables.
>       access_key                  = "s3 user access key"
>       secret_key                  = "s3 user secret key"  
> ```
> 
> WDYT?

I changed for high perf bucket

**Review by scraly — CHANGES_REQUESTED** [review]

**Review by scraly — CHANGES_REQUESTED** [review]
There is a blocking error during the terraform plan command execution.
The AWS provider can't be initialized because credentials are not given (we can't define a parameter with a resource that will be created after).

File: use-cases/kubeflow/README.md [hunk 23]
```diff
@@ -0,0 +1,133 @@
...
+
+After a while the status of the run should turn green. You can see the logs of the run in your object storage.
+
+## Troubleshoot
+
+### Access the Kubernetes cluster
+
+```bash
+terraform output --raw ovh_kube_cluster_kubeconfig > ./kubeconfig
+export KUBECONFIG=./kubeconfig
```
Comment (scraly) [inline use-cases/kubeflow/README.md:131]:
```suggestion
```

File: use-cases/kubeflow/README.md [hunk 24]
```diff
@@ -0,0 +1,133 @@
...
+kubectl get nodes
```
Comment (scraly) [inline use-cases/kubeflow/README.md:132]:
```suggestion
kubectl get nodes --kubeconfig ./kubeconfig
```

**Review by scraly — CHANGES_REQUESTED** [review]
Remarks for troubleshooting in kubernetes

**Comment by scraly** [comment]
FYI There is a usage of a deprecated resource:
image

**Review by scraly — CHANGES_REQUESTED** [review]
@sguyennet ping me again when wyou'll have time to fix last changes requests :)

**Comment by sguyennet** [comment]
> FYI There is a usage of a deprecated resource: image

This is fixed

File: use-cases/kubeflow/README.md [hunk 25]
```diff
@@ -5,8 +5,9 @@ The purpose of this tutorial is to deploy KubeFlow in an OVHcloud Managed Kubern
 This Terraform will create and configure:
 
 * A private network
+* A gateway
 * A managed Kubernetes cluster
-* A load balancer for Kubernetes with a public IP
+* A load balancer with a public IP
```
Comment (scraly) [inline use-cases/kubeflow/README.md:10]: A Public Cloud Load Balancer

**Review by scraly — CHANGES_REQUESTED** [review]

**Pull request merged**